File: cw_funkcje02.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  cw_funkcje02.py
#  
#  

import math
import statistics
def srednia(tablica, ilosc ilczb):
	srednia = sum(tablica)/ilosc liczb
	print("Średnia wynosi: ", srednia)
	
def mediana(oceny):
	oceny = sorted(oceny) # sortowanie rosnące liczb
	ile = len(oceny)
	if ile % 2 == 0:
		indeks = int(ile / 2)
		mediana = oceny[indeks] + oceny[indeks - 1]
	else:
		indeks =
		mediana = ocen[int(ile / 2)]
	print("Mediana wynosi: ", statistics.mediana(tablica))
	
def main(args):
	ile = 20
	oceny = []
	ocenyile = [0,0,0,0,0,0,0]
	while ile > 0:
		ocena = int(input("Podaj ocene lub zero, aby zakończyć: "))
		if ocena > 0 and ocena < 7:
			oceny.append(ocena)
			ocenyile[oceny]
	ilosc liczb = int(input("Ilość ocen: "))
	if ilosc liczb > 20:
		print("Maksymalna ilość ocen wynosi 20")
		return(ilosc liczb)
		
	tablica = {}
	for i in range(0, ilosc liczb):
		f = int(input("Podaj ocenę: "))
		tablica.append (f)
		
		if  f <= 0 or f > 6:
			print("Wprowadziłeś/aś niepoprawną ocenę!")
			
		return
		
	srednia(tablica, ilosc liczb)
	mediana(tablica)
	
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cw_funkcje02.cpp
/*
 * cw_funkcje02.cpp
 * 
 */

#include <iostream>
using namespace std;


void srednia(float suma, float ilosc liczb) {
	float srednia = suma / ilosc liczb;
	cout << "srednia wynosi: " << srednia << endl;
}

void mediana(float * tab, int ilosc liczb) {
	float mediana;
	if(ilosc liczb % 2 == 0) {
		mediana = ( tab[ilosc liczb /2 - 1] + tab[ilosc liczb /2]) /2;
	} else {
		mediana = tab[ilosc liczb /2] ;
	}
	cout <<"Mediana wynosi: " << mediana << endl;
}

int main(int argc, char **argv)
{
	int t = 0;
	int ilosc liczb = 0;
	int * tab = new int [ilosc liczb];
	cout << "Ilość liczb: ";
	cin >> ilosc liczb;
	
	if(ilosc liczb > 20) {
		cout <<"Maksymalna ilość ocen wynosi 20" << endl;
	
	return 0;

	float tablica[ilosc liczb];
	int suma = 0:
	for (int i = 0; i < ilosc liczb; ++i)
{
		cout << "Podaj ocenę: " <<(i + 1) << " : ";
		cin >> tablica [i];
		if(tablica [i] < 1 || tablica [i] > 6) {
			cout << "Wprowadziłeś/aś niepoprawną ocenę!" << endl;
			return 0
	}
		suma += tablica[i]
}
	int v;
	int x;
	for(v = 0; v < ilosc liczb; ++v)
	{
				for(x = v + 1; x < ilosc liczb; ++x)
		{
			if(tablica [v] > tablica [x])
			{
				t = tablica [v]
				tablica [v] = tablica [x];
				t = tablica [x];
			}
		}
	}
	srednia (suma, ilosc liczb);
	mediana (tablica, ilosc liczb);
	
	
	return 0;
}
